lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -129,8 +147,101 @@ data class GetTransactionStatusResponse(
 
     @SerialName("execution_status")
     val executionStatus: TransactionExecutionStatus? = null,
+
+    @SerialName("failure_reason")
+    val failureReason: String? = null,
 ) : StarknetResponse
 
+@Serializable
+data class MessageStatus(
+    @SerialName("transaction_hash")
+    val transactionHash: Felt,
+
+    @SerialName("finality_status")
+    val finalityStatus: TransactionStatus,
+
+    @SerialName("failure_reason")
+    val failureReason: String? = null,
+)
+
+@Serializable
+data class StorageProof(
+    @SerialName("classes_proof")
+    val classesProof: NodeHashToNodeMapping,
+
+    @SerialName("contracts_proof")
+    val contractsProof: ContractsProof,
+
+    @SerialName("contracts_storage_proofs")
+    val contractsStorageProofs: List<NodeHashToNodeMapping>,
+
+    @SerialName("global_roots")
+    val globalRoots: GlobalRoots,
+) : StarknetResponse {
+    @Serializable
+    data class GlobalRoots(
+        @SerialName("contracts_tree_root")
+        val contractsTreeRoot: Felt,
+
+        @SerialName("classes_tree_root")
+        val classesTreeRoot: Felt,
+
+        @SerialName("block_hash")
+        val blockHash: Felt,
+    )
+}
+
+@Serializable
+data class ContractsProof(
+    @SerialName("nodes")
+    val nodes: NodeHashToNodeMapping,
+
+    @SerialName("contract_leaves_data")
+    val contractLeavesData: List<ContractLeafData>,
+)
+
+@Serializable
+data class ContractLeafData(
+    @SerialName("nonce")
+    val nonce: Felt,
+
+    @SerialName("class_hash")
+    val classHash: Felt,
+)
+
+@Serializable
+data class NodeHashToNodeMappingItem(
+    @SerialName("node_hash")
+    val nodeHash: Felt,
+
+    @SerialName("node")
+    val node: MerkleNode,
+) {
+    @Serializable
+    sealed interface MerkleNode
```

Comment:
Are you sure that works?

Seems to me like some custom serializer is needed for `node` 🤔

Reply:
I added missing `MerkleNodeSerializer` 👍 .